src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonUtilities.Schema.cs

```diff
@@ -76,6 +77,22 @@ public static JsonElement CreateFunctionJsonSchema(
                 Throw.ArgumentException(nameof(parameter), "Parameter is missing a name.");
             }
 
+            if (parameter.ParameterType == typeof(CancellationToken))
+            {
+                // CancellationToken is a special case that, by convention, we don't want to include in the schema.
+                // Invocations of methods that include a CancellationToken argument should also special-case CancellationToken
+                // to pass along what relevant token into the method's invocation.
+                continue;
+            }
+
+            if (parameter.GetCustomAttribute<FromServiceProviderAttribute>(inherit: true) is not null)
```

Comment:
This is introducing a concern that is specific to AIFunctionFactory infrastructure. I wonder if it's further evidence that the two components should be merged as one.

Reply:
Note that this PR also highlights that AIFunctionFactory should not be moved down. It needs to reference IKeyedServiceProvider, which is in the DI assembly, and we won't be adding the DI assembly as a dependency of the abstractions lib.